trade-finance-manager-ui/server/controllers/case/parties/index.js

```diff
@@ -186,31 +246,154 @@ const postPartyDetails = (partyType) => (
       }
     }
 
+    // Fetches company information from URN
+    const company = await api.getParty(partyUrn);
+
+    // Non-existent party urn
+    if (!company && !company.data) {
+      return res.render('case/parties/void.njk', {
+        dealId,
+        party,
+      });
+    }
+
+    // Agent party only - Commission rate
+    if (req.body.commissionRate) {
+      req.session.commissionRate = req.body.commissionRate;
+    }
+
+    // Redirect to summary (confirmation) page
+    return res.redirect(`/case/${dealId}/parties/${party}/summary/${partyUrn}`);
+  } catch (e) {
+    console.error('Error posting party URN summary page', { e });
+    return res.redirect('/not-found');
+  }
+};
+
+/**
+ * Submits confirmed party URN to the TFM
+ * @param {Express.Request} req
+ * @param {Express.Response} res
+ * @returns {Object} Express response, if successful then re-directed to all parties page
+ */
+const postPartyDetails = async (req, res) => {
+  try {
+    const { user } = req.session;
+
+    delete req.body._csrf;
```

Comment:
Not a blocker as PR is urgent - do we need to delete this?

Reply:
yeah. it should be in the payload otherwise it ends up in the db